tests/Functional/EventListener/UserContextListenerTest.php

```diff
@@ -11,34 +11,41 @@
 
 namespace FOS\HttpCacheBundle\Tests\Functional\EventListener;
 
+use FOS\HttpCacheBundle\Tests\Functional\SessionHelperTrait;
 use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
-use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
 
 class UserContextListenerTest extends WebTestCase
 {
+    use SessionHelperTrait;
+
     /**
      * @dataProvider userHashDataProvider
      */
     public function testHashLookup(string $username, string $hash)
     {
+        // as we tamper with the session id, make sure no previous session is around
+        @unlink('/tmp/test.mocksess');
+
         $client = static::createClient([], $username ? [
             'PHP_AUTH_USER' => $username,
             'PHP_AUTH_PW' => $username,
         ] : []);
-        /** @var SessionInterface $session */
-        $session = $client->getContainer()->get('session');
-        $session->setId('test');
-        $session->start();
+
+        $this->callInRequestContext($client, [$this, 'setSessionId']);
+
+        // remove basic auth to avoid restarting the session
+        $client->setServerParameters([]);
```

Comment:
okay, this at least works around the issue. not sure if it is supposed to happen or not (session gets migrated on every request when basic auth headers are sent in request).

Reply:
thanks. i asked wouter on symfony slack what the idea with basic auth & session is. (its a weird edge case anyways, for our tests i am happy with this solution)